drivers/clock_control/clock_stm32_ll_common.c

```diff
@@ -101,7 +101,9 @@ static inline int stm32_clock_control_on(const struct device *dev,
 	defined(CONFIG_SOC_SERIES_STM32WBX) || \
 	defined(CONFIG_SOC_SERIES_STM32G4X)
 	case STM32_CLOCK_BUS_AHB2:
+	#ifdef RCC_AHB2_SUPPORT
```

Comment:
Please move to a dedicated commit.
Then, `RCC_AHB2_SUPPORT` is defined only on F4, so this works in your case, but will break other cases.
I propose the folllowing:
```
#if defined(CONFIG_SOC_SERIES_STM32L4X) || \
	defined(CONFIG_SOC_SERIES_STM32L5X) || \
	defined(CONFIG_SOC_SERIES_STM32F4X) && !defined(CONFIG_SOC_STM32F410RX) || \
	defined(CONFIG_SOC_SERIES_STM32F7X) || \
	defined(CONFIG_SOC_SERIES_STM32F2X) || \
	defined(CONFIG_SOC_SERIES_STM32WBX) || \
	defined(CONFIG_SOC_SERIES_STM32G4X)
        case STM32_CLOCK_BUS_AHB2:
        [...]
```